torchdata/datapipes/utils/common.py

```diff
@@ -22,3 +22,13 @@ def validate_pathname_binary_tuple(data: Tuple[str, IOBase]):
             f"binary stream within the tuple should have IOBase or"
             f"its subclasses as type, but it is type {type(data[1])}"
         )
+
+
+def _no_op_fn(*args):
+    """
+    No-operation function, returns passed arguments , always as iterable.
+    """
+    if len(args) == 1 and isinstance(args[0], (list, tuple)):
+        return args[0]
+    else:
+        return args
```

Comment:
```suggestion
    if len(args) == 1:
        try:
            return iter(args[0])
        except TypeError:
            pass
    return args
```
@mobikoz What do you think of this instead? This will allow objects that are not `list` or `tuple` to be unpacked (e.g. `numpy.array`).

cc: @ejguan

Reply:
Can I know why do we want to differentiate the cases between size 1 and other sizes?
```py
list(dp)  # [[1], [2], [3], [4],]
list(dp.flatmap())  # I would expect the result as [1, 2, 3, 4]
```

BTW, if this function is not commonly used by other DataPipes, we can put it in the `callable.py` file.

Reply:
> Can I know why do we want to differentiate the cases between size 1 and other sizes?

As far as I observe, because the input to the function is `*args`, there are two main cases:
1. `input_col=None`
2. `input_col` only specifies one column

Reply:
@ejguan   This differentiation is the consequence of using variable length *args for the sake of input_col parameter use.
If not handled, in case of given example: 
'
list(dp)  # [[1], [2], [3], [4],]
list(dp.flatmap())  #  result will be [[1], [2], [3], [4],]
'
This is because args would be always of type tuple, in this case for eg. ([1],) for the first element.

Reply:
I see, makes sense. Thanks for the explanation.

Reply:
I guess the only thing that is does is that if `arg[0]` isn't iterable (can't be unpacked), it returns `args`.

But in this case that is a kind of silent failure, which I think is worse. So, my bad. Let's do this instead:
```python
    if len(args) == 1:
        return args[0]
    return args
```

Sorry about this @mobikoz. After this, we should be about to merge the change. Thanks!

Reply:
I believe that currently committed implementation is correct - situation when arg[0] is not iterable seems to be common and TypeError will be rised. 

Consider example
`list(dp) # [[1, 2, 3], [4, 5, 6]]`
`list(dp.flatmap(input_col=1)) # TypeError will be raised`

Reply:
My point is that the `TyperError` should be raised for your example (but it doesn't in the current implementation because it catches the exception). 
Otherwise, it would be a silent failure, because the DataPipe promises to unpack the element but it actually didn't and just returned the element itself.

Reply:
Currently it will work like that:
`list(dp) # [[1, 2, 3], [4, 5, 6]]`
`list(dp.flatmap(input_col=1) # [2, 5]`


`list(dp) # [1, 2, 3, 4, 5, 6]`
`list(dp.flatmap(input_col=1) # [1, 2, 3, 4, 5, 6]`

Isn't it expected behavior ? 
Just double checking if I understand correctly.

Reply:
btw it is analogous behavior to calling flatmap with some fn, for eg 
`def fn(e):`
`    return` [e, e * 10]`

Reply:
I might be wrong but you might want to do this if all you want is to select an index?

```python
from operator import itemgetter
dp = IterableWrapper([[1, 2, 3], [4, 5, 6]]).map(itemgetter(1))
list(dp)  # [2, 5]
```

The case I worry about is something like this:
```python
dp = IterableWrapper([[1, [2, 2], 3], [4, [5, 5], 6], [7, 8, 9], [10, [11, 11], 12]]).flatmap(input_col=1)
```
The 3rd list will silently unpack to `8` even though it is different from the other lists.

@ejguan any thoughts on which behavior is better?

Reply:
How about this?
```py
list(dp) # [[1, [2], 3], [4, [5], 6]]
list(dp.flatmap(input_col=1) # I expect [2, 5] rather than [[2], [5]]
```

I feel like it's fine to remove `iter` from the currently implementation as long as we make it clear in the comment that we would only allow the output being unpackable by `yield from`

Reply:
Pushed corrected implementation